tests/unit/test_matcher.c

```diff
@@ -337,23 +339,22 @@ Test(matcher, test_replace_works_on_matches_input_with_store_matches_too)
   LogMessage *msg = log_msg_new_empty();
   log_msg_set_value_by_name(msg, "1", test_message, -1);
 
-  GlobalConfig *cfg = cfg_new_snippet();
-  replace_template = log_template_new(cfg, NULL);
+  replace_template = log_template_new(configuration, NULL);
   cr_assert(log_template_compile(replace_template, replace_pattern, NULL));
 
   NVHandle input_handle = log_msg_get_value_handle("1");
   const gchar *input = log_msg_get_value(msg, input_handle, &value_len);
 
+  NVTable *payload = nv_table_ref(msg->payload);
   const gchar *result = log_matcher_replace(m, msg, input_handle, input, value_len,
                                             replace_template, &result_len);
+  nv_table_unref(payload);
   cr_log_info("replace result value: %s, length(%ld)", result, result_len);
   cr_assert_arr_eq(result, expected_result, strlen(expected_result),
-                   "replace failed; result: %s (length %ld), expected: %s (length %d)",
+                   "replace failed; result: %s (length %ld), expected: %s (length %ld)",
                    result, result_len, expected_result, strlen(expected_result));
 
-  scratch_buffers_reclaim_allocations();
   log_template_unref(replace_template);
   log_matcher_unref(m);
   log_msg_unref(msg);
-  cfg_free(cfg);
```

Comment:
shouldn't this move to the teardown() function as well?

Reply:
done.